packages/flutter/test/material/app_bar_theme_test.dart

```diff
@@ -182,6 +182,43 @@ void main() {
     // Default value for ThemeData.typography is Typography.material2014()
     expect(text.style, Typography.material2014().englishLike.bodyText2.merge(Typography.material2014().white.bodyText2).merge(themeData.primaryTextTheme.bodyText2));
   });
+
+  testWidgets('AppBar uses AppBarTheme.centerTitle when centerTitle is null', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(appBarTheme: const AppBarTheme(centerTitle: true)),
+      home: Scaffold(appBar: AppBar(title: const Text('Title'))),
+    ));
+
+    final NavigationToolbar navToolBar = tester.widget(find.byType(NavigationToolbar));
+    expect(navToolBar.centerMiddle, true);
```

Comment:
Could you write these tests to be similar to [this example](https://github.com/flutter/flutter/blob/51cf4b327850042c313ed356cc2dbdac91b18d64/packages/flutter/test/material/app_bar_test.dart#L151)?

It's preferable to check for the locations of the titles rather than checking that a property is `true`/`false`. The concern here being that there could be a bugs where `centerMiddle` is set to `true`, but the render object decides to place itself in a non-center location.

Reply:
I had written a similar test, but then I thought it might not be relevant cause `AppBarTheme` is only responsible for providing the value and not rendering the widget. WDYT?

Reply:
I think your reasoning makes sense. I just posed the question in the `hackers-framework` channel of the `flutter` Discord channel to see if anybody has any thoughts about this. I don't really have a good answer except that I felt that checking the rendered location felt "safer" :)

Reply:
Sure, I'll follow up 😃

Reply:
The existing test should be good to go!